src/components/forms/ProxyDeviceForm.tsx

```diff
@@ -0,0 +1,303 @@
+import { FC } from "react";
+import {
+  Button,
+  Icon,
+  Input,
+  Label,
+  useNotify,
+} from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { LxdDeviceValue, LxdProxyDevice } from "types/device";
+import { InstanceAndProfileFormikProps } from "./instanceAndProfileFormValues";
+import { fetchProfiles } from "api/profiles";
+import { getInheritedProxies } from "util/configInheritance";
+import Loader from "components/Loader";
+import { ProxyCard } from "types/resources";
+import ScrollableForm from "components/ScrollableForm";
+import RenameDeviceInput from "components/forms/RenameDeviceInput";
+import ConfigurationTable from "components/ConfigurationTable";
+import { MainTableRow } from "@canonical/react-components/dist/components/MainTable/MainTable";
+import { getConfigurationRowBase } from "components/ConfigurationRow";
+import classnames from "classnames";
+import {
+  addNoneDevice,
+  deduplicateName,
+  findNoneDeviceIndex,
+  removeDevice,
+} from "util/formDevices";
+import { getInheritedDeviceRow } from "components/forms/InheritedDeviceRow";
+import { deviceKeyToLabel } from "util/devices";
+import { ensureEditMode } from "util/instanceEdit";
+import SelectProxyBtn from "components/forms/SelectProxyBtn";
+// import ProxyDeviceInput from "./ProxyDeviceInput";
+
+interface Props {
+  formik: InstanceAndProfileFormikProps;
+  project: string;
+}
+
+const ProxyDeviceForm: FC<Props> = ({ formik, project }) => {
+  const notify = useNotify();
+
+  const {
+    data: profiles = [],
+    isLoading: isProfileLoading,
+    error: profileError,
+  } = useQuery({
+    queryKey: [queryKeys.profiles],
+    queryFn: () => fetchProfiles(project),
+  });
+
+  if (profileError) {
+    notify.failure("Loading profiles failed", profileError);
+  }
+
+  const inheritedProxies = getInheritedProxies(formik.values, profiles);
+
+  const existingDeviceNames: string[] = [];
+  existingDeviceNames.push(...inheritedProxies.map((item) => item.key));
+  existingDeviceNames.push(...formik.values.devices.map((item) => item.name));
+
+  const addProxyCard = (card: ProxyCard) => {
+    const copy = [...formik.values.devices];
+    copy.push({
+      type: "proxy",
+      bind: card.bind,
+      nat: card.nat,
+      proxy_protocol: card.proxy_protocol,
+      mode: card.listen?.mode,
+      listen: `${card.listen?.type}:${card.listen?.address}:${card.listen?.port}`,
+      connect: `${card.listen?.type}:${card.listen?.address}:${card.listen?.port}`,
```

Comment:
```suggestion
      type: "proxy",
```
I think we can keep the optional fields blank on creation

Reply:
For some reason, if I do not put the fields here, then I cannot render the labels, but I can also no longer put the fields here anyway because upon implementing the suggestion to get rid of the card, I can no longer reference the card attributes.

This ProxyCard that is created in the AddProxy function -to my understanding - contains the attributes that will be shown on the proxydevice form. This blank slate if you will, is then pushed onto the devices stack, and later re-referenced within Object.keys at line 193 and getConfigurationRowBase around line 201 and iterated through to identify how to render each component. I don't think we can just get rid of the proxy card.